### PR TITLE
unify doc preview URLs

### LINKED ISSUE #3894: Unify HUD and docs preview paths?
We just enabled docs preview on TorchVision by uploading it to docs-preview.pytorch.org. However, we noticed that the paths diverge from the ones used by the HUD. #3891 was a quick fix to get it up and running, but we should have a discussion around unifying this.

## HUD

- https://hud.pytorch.org/hud/pytorch/pytorch/master
- https://hud.pytorch.org/hud/pytorch/pytorch/nightly
- https://hud.pytorch.org/hud/pytorch/vision/main
- https://hud.pytorch.org/hud/pytorch/audio/main

scheme: `https://hud.pytorch.org/hud/$OWNER/$REPO/$BRANCH`

## HUD linked from bot PR comments

- https://hud.pytorch.org/pr/96954 (this is what the bot comments, but it seems to be shortcut for the link below)
- https://hud.pytorch.org/pr/pytorch/pytorch/96131
- https://hud.pytorch.org/pr/pytorch/vision/7427

scheme: `https://hud.pytorch.org/pr/$OWNER/$REPO/$PR`

## Doc preview linked from bot comments

- https://docs-preview.pytorch.org/96131/index.html
- https://docs-preview.pytorch.org/96131/cppdocs/index.html
- https://docs-preview.pytorch.org/vision/7427/index.html

There is no unifying scheme. PyTorch core uses `https://docs-preview.pytorch.org/$PR/index.html` whereas TorchVision uses `https://docs-preview.pytorch.org/$REPO/$PR/index.html`. Both diverge from the HUD links in the sense that `$OWNER` is never used.

## Proposal

Docs preview should be served under `https://docs-preview.pytorch.org/$OWNER/$REPO/$PR/index.html` to align with the HUD links. This should not break any pre-existing links, since the proposed scheme can coexist with the old one.

### PR DESCRIPTION
**Pull request by pmeier**
Closes #3894. This needs to be merged in unison with a PR on PyTorch core and torchvision. I'll link them here when they are up.

### CODE REVIEW

File: torchci/lib/drciUtils.ts [hunk 1]
```diff
@@ -10,9 +10,9 @@ export const OWNER: string = "pytorch";
 export const DRCI_COMMENT_START = "<!-- drci-comment-start -->\n";
 export const OH_URL =
     "https://github.com/pytorch/pytorch/wiki/Dev-Infra-Office-Hours";
-export const DOCS_URL = "https://docs-preview.pytorch.org/";
-export const PYTHON_DOCS_URL = "/index.html";
-export const CPP_DOCS_URL = "/cppdocs/index.html";
+export const DOCS_URL = "https://docs-preview.pytorch.org";
+export const PYTHON_DOCS_PATH = "index.html";
+export const CPP_DOCS_PATH = "cppdocs/index.html";
```
Comment (pmeier) [inline torchci/lib/drciUtils.ts:15]: I've removed the leading and trailing `/` here, because I found it rather hard to read below when I needed a `/`.

**Comment by pmeier** [comment]
- PyTorch: pytorch/pytorch#97433
- TorchVision: we are already using the [`${OWNER}/${SCHEME}`](https://github.com/pytorch/vision/blob/18a2e8eb5c6e30e2bc22416379b10f5dfaccc4d4/.github/workflows/docs.yml#L68). Thus, we don't need to do anything there and only fix our CloudFront setup.

**Review by seemethere — APPROVED** [review]

**Review by malfet — APPROVED** [review]
Please plan carefully how you stage those changes, in order not to break folks with their work in progress.

**Review by osalpekar — APPROVED** [review]

**Comment by pmeier** [comment]
Both the PyTorch and TorchVision PR have been merged. Meaning, we can move ahead with this PR in a week or so to have a proper grace period.

**Pull request merged**